tutor/src/components/course-calendar/add.cjsx

```diff
@@ -38,6 +41,19 @@ CourseAdd = React.createClass
       open: false
     })
 
+  getDateType: ->
+    {referenceDate, addDate} = @state
+    {termStart, termEnd} = @props
+
+    if addDate?.isBefore(termStart, 'day')
+      'day before term starts'
+    else if addDate?.isAfter(termEnd, 'day')
+      'day after term ends'
+    else if addDate?.isBefore(referenceDate, 'day')
+      'past day'
+    else if addDate?.isSame(referenceDate, 'day')
+      'today'
+
```

Comment:
if addDate is null because state hasn't set it, then there's no message returned.  Is that OK?

If so, we could do an early `return '' unless addDate?` and then wouldn't need the `?` on each line.   Could use a `switch` statement as well, but that's up to you, I'm kinda ambivalent about wether they're better than `if`